openslides_gui/gui.py

```diff
@@ -0,0 +1,763 @@
+        self._update_interval_choices()
+
+    @property
+    def interval_unit(self):
+        return self._INTERVAL_UNITS[self.ch_interval_unit.GetSelection()]
+
+    @interval_unit.setter
+    def interval_unit(self, unit):
+        try:
+            idx = self._INTERVAL_UNITS.index(unit)
+        except IndexError:
+            raise ValueError("Unknown unit {0}".format(unit))
+
+        self.ch_interval_unit.SetSelection(idx)
+
+    def _update_interval_choices(self):
+        count = self.sb_interval.GetValue()
+        choices = [
+            ngettext("second", "seconds", count),
+            ngettext("minute", "minutes", count),
+            ngettext("hour", "hours", count),
+        ]
+
+        current = self.ch_interval_unit.GetSelection()
+        if current == wx.NOT_FOUND:
+            current = 2  # default to hour
+
+        self.ch_interval_unit.Clear()
+        self.ch_interval_unit.AppendItems(choices)
+        self.ch_interval_unit.SetSelection(current)
+
+    def _update_backup_enabled(self):
+        checked = self.cb_backup.IsChecked()
+        self.fp_dest.Enable(checked)
+        self.sb_interval.Enable(checked)
+        self.ch_interval_unit.Enable(checked)
+
+    def on_backup_checked(self, evt):
+        self._update_backup_enabled()
+
+    def on_interval_changed(self, evt):
+        self._update_interval_choices()
+
+    # TODO: validate settings on close (e.g. non-empty path if backup is
+    #       enabled)
+
+
+class MainWindow(wx.Frame):
+    def __init__(self, parent=None):
+        super(MainWindow, self).__init__(parent, title="OpenSlides")
+        icons = wx.IconBundle(
+            get_data_path("openslides.ico"),
+            wx.BITMAP_TYPE_ICO)
+        self.SetIcons(icons)
+
+        self.server_running = False
+
+        self.gui_settings_path = None
+        self.gui_initialized = False
+
+        self.backupdb_enabled = False
+        self.backupdb_destination = ""
+        self.backupdb_interval = 15
+        self.backupdb_interval_unit = "minute"
+        self.last_backup = None
+
+        self.backup_timer = wx.Timer(self)
+        self.Bind(wx.EVT_TIMER, self.on_backup_timer, self.backup_timer)
+
+        spacing = 5
+
+        panel = wx.Panel(self)
+        grid = wx.GridBagSizer(spacing, spacing)
+
+        # logo & about button
+        logo_box = wx.BoxSizer(wx.HORIZONTAL)
+        grid.Add(logo_box, pos=(0, 0), flag=wx.EXPAND)
+        row = 0
+
+        fp = get_data_path("openslides-logo_wide.png")
+        with open(fp, "rb") as f:
+            logo_wide_bmp = wx.Image(f).ConvertToBitmap()
+
+        logo_wide = wx.StaticBitmap(panel, wx.ID_ANY, logo_wide_bmp)
+        logo_box.AddSpacer(2 * spacing)
+        logo_box.Add(logo_wide)
+        logo_box.AddStretchSpacer()
+
+        # TODO: get_version was removed - replace with new method
+         #version_str = _("Version {0}").format(openslides.get_version())
+        version_str = _("Version {0}").format("???")
```

Comment:
Please use for version_str:
```python
import openslides
print(openslides.__version__)
```
see https://github.com/OpenSlides/OpenSlides/issues/1492#issuecomment-99356146